packages/itwin/tree-widget/src/components/trees/models-tree/internal/ModelsTreeVisibilityHandler.ts

```diff
@@ -154,24 +157,56 @@ class ModelsTreeVisibilityHandlerImpl implements HierarchyVisibilityHandler {
   }
 
   public async getVisibilityStatus(node: HierarchyNode): Promise<VisibilityStatus> {
-    return firstValueFrom(this.getVisibilityStatusObs(node));
+    return firstValueFrom(
+      this.getVisibilityStatusObs(node).pipe(
+        // unsubscribe from the observable if the change request for this or ancestor node is received
+        takeUntil(this._changeRequest.pipe(filter(({ key, depth }) => depth <= node.parentKeys.length || HierarchyNodeKey.equals(node.key, key)))),
```

Comment:
Not sure the `depth <= node.parentKeys.length` is enough to say that it's an ancestor. E.g., we could have this:

```
- A
  - B
- C
  - D
```

We may be checking status of `B` and changing status of `C`. Depth of `C` is smaller than of `B`, but it's not an ancestor.